instrumentation/koala/lib/opentelemetry/instrumentation/koala/patches/instrumentation.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Koala
+      module Patches
+        # Module to prepend to Koala::Facebook::API for instrumentation
+        module Instrumentation
+          VERBS_TO_SPAN_NAMES = Hash.new { |h, k| h[k] = "Koala #{k}" }
+
+          def graph_call(path, args = {}, verb = 'get', options = {}, &post_processing)
+            tracer.in_span(
+              VERBS_TO_SPAN_NAMES[req.method],
+              {
+                'koala.verb' => verb,
+                'koala.path' => path
+              },
+              kind: :client
+            ) do |span|
+              rescue_api_error(span) { super(path, args, verb, options, &post_processing) }
+            end
+          end
+
+          private
+
+          def annotate_span_with_api_error_rescue!(span, &block)
+            block.call
+          rescue Koala::Facebook::APIError => e
+            fb_error_trace_id = e&.fb_error_trace_id
+            fb_error_type = e&.fb_error_type
+            fb_error_code = e&.fb_error_code
+            fb_error_subcode = e&.fb_error_code
+
+            span.set_attribute('koala.fb_error_trace_id', fb_error_trace_id)
+            span.set_attribute('koala.fb_error_type', fb_error_type)
+            span.set_attribute('koala.fb_error_code', fb_error_code)
+            span.set_attribute('koala.fb_error_subcode', fb_error_subcode)
+            raise
```

Comment:
https://github.com/SomalianIvan/opentelemetry-ruby/blob/23e1ba358f5e9c95f6ff898e1aca98b25e99b55a/api/lib/opentelemetry/trace/tracer.rb#L30-L37 - this should `finish` the span